Classes/Context/TopwireContextFactory.php

```diff
@@ -3,42 +3,38 @@
 namespace Topwire\Context;
 
 use Psr\Http\Message\ServerRequestInterface;
+use Symfony\Component\DependencyInjection\Attribute\Autoconfigure;
 use Topwire\Context\Attribute\Plugin;
+use Topwire\TopwireException;
 use TYPO3\CMS\Core\Utility\GeneralUtility;
 use TYPO3\CMS\Core\Utility\MathUtility;
 use TYPO3\CMS\Extbase\Configuration\ConfigurationManagerInterface;
 use TYPO3\CMS\Extbase\Service\ExtensionService;
-use TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController;
+use TYPO3\CMS\Extbase\Utility\ExtensionUtility;
 
 class TopwireContextFactory
 {
-    private TypoScriptFrontendController $typoScriptFrontendController;
-
-    public function __construct(TypoScriptFrontendController $typoScriptFrontendController)
-    {
-        $this->typoScriptFrontendController = $typoScriptFrontendController;
-    }
+    public function __construct(private readonly ServerRequestInterface $request)
+    {}
 
     /**
      * @param array<string, mixed> $arguments
      */
     public function forRequest(
-        ServerRequestInterface $request,
         array $arguments,
         ?ConfigurationManagerInterface $configurationManager = null,
     ): TopwireContext {
-        $extensionName = $arguments['extensionName'] ?? $request->getAttribute('extbase')?->getControllerExtensionName();
-        $pluginName = $arguments['pluginName'] ?? $request->getAttribute('extbase')?->getPluginName();
+        $extensionName = $arguments['extensionName'] ?? $this->request->getAttribute('extbase')?->getControllerExtensionName();
+        $pluginName = $arguments['pluginName'] ?? $this->request->getAttribute('extbase')?->getPluginName();
         $actionName = $arguments['action'] ?? null;
-        $configurationManager ??= GeneralUtility::makeInstance(ConfigurationManagerInterface::class);
-        $extensionService = new ExtensionService();
-        $extensionService->injectConfigurationManager($configurationManager);
-        $pluginNamespace = $extensionService->getPluginNamespace($extensionName, $pluginName);
+        $pluginNamespace = GeneralUtility::makeInstance(ExtensionService::class)->getPluginNamespace($extensionName, $pluginName);
 
         // @todo: decide whether this needs to be changed, or set via argument, or maybe even removed completely
         $isOverride = isset($arguments['extensionName']);
-        $contentRecordId = $isOverride ? null : $configurationManager->getContentObject()?->currentRecord;
-
+        $contentRecordId = $isOverride ? null : $this->request->getAttribute('currentContentObject')->data['uid'] ?? null;
```

Comment:
This is wrong here. At this point, it should be $this->request->getAttribute('currentContentObject')->currentRecord
Otherwise, only the UID is returned here without reference to the table (i.e. tt_content:215), which leads to incorrect behavior in the function resolveContextRecord (where "pages" is always used as the table).

Reply:
True, thank you!